client/my-sites/plans/current-plan/jetpack-checklist/index.tsx

```diff
@@ -68,7 +69,7 @@ class JetpackChecklist extends PureComponent< Props > {
 	}
 
 	isComplete( taskId: string ): boolean {
-		return get( this.props.taskStatuses, [ taskId, 'completed' ], false );
+		return getTaskList( this.props ).isCompleted( taskId );
```

Comment:
I don't have much understanding or intuition around `getTaskList` which makes me a bit hesitant to shove task data into it for the Jetpack checklist.

I would like to see convergence in the way our checklists approach data, but moving the Jetpack checklist to the task list object here in order to handle the different shape of data from the API feels a bit rushed. What are the tradeoffs we're making?

How do you feel about making the Jetpack checklist handle the different shape of data without requiring getTaskList?

```patch
diff --git a/client/my-sites/plans/current-plan/jetpack-checklist/index.tsx b/client/my-sites/plans/current-plan/jetpack-checklist/index.tsx
index b49e2e5559..5e498ff86c 100644
--- a/client/my-sites/plans/current-plan/jetpack-checklist/index.tsx
+++ b/client/my-sites/plans/current-plan/jetpack-checklist/index.tsx
@@ -47,11 +47,10 @@ import withTrackingTool from 'lib/analytics/with-tracking-tool';
 import './style.scss';
 import { getTaskList } from 'my-sites/checklist/wpcom-checklist/wpcom-task-list';
 
-interface ChecklistTaskState {
-	[taskId: string]: {
-		completed: null | boolean;
-	};
-}
+type ChecklistTaskState = Array< {
+	id: string;
+	isCompleted: boolean;
+} >;
 
 interface Props {
 	isPremium: boolean;
@@ -69,7 +68,11 @@ class JetpackChecklist extends PureComponent< Props > {
 	}
 
 	isComplete( taskId: string ): boolean {
-		return getTaskList( this.props ).isCompleted( taskId );
+		if ( this.props.taskStatuses ) {
+			const task = this.props.taskStatuses.find( ( { id } ) => id === taskId );
+			return task ? task.isCompleted : false;
+		}
+		return false;
 	}
 
 	handleTaskStart = ( { taskId, tourId }: { taskId: string; tourId?: string } ) => () => {
```